README.md

```diff
@@ -5,22 +5,295 @@
+    ```
+
+## Samples
+
+For complete samples check the [Realm Flutter and Dart Samples](https://github.com/realm/realm-dart-samples).
+
+## Documentation
+
+For API documentation go to 
+ * [Realm Flutter API Docs](https://pub.dev/documentation/realm/latest/)
+
+ * [Realm Dart API Docs](https://pub.dev/documentation/realm_dart/latest/)
+
+For a complete documentation go to [Realm Flutter and Dart SDK Docs](https://docs.mongodb.com/realm/sdk/flutter/).
+
+
+## Limitations
+
+* This version of Realm Flutter and Dart SDK allows working with a local only (on device) Realm database in Flutter and Dart desktop. Realm Sync functionality is not implemented.
+
+* It provides the functionality for creating, retrieving, querying, sorting, filtering, updating Realm objects and supports change notifications.
+
+* Flutter Desktop on Linux is not yet support
+
+# Realm Flutter SDK 
+
+The Realm Flutter package name is `realm`
+
+## Environment setup for Realm Flutter
+
+* Supported platforms are Flutter (iOS, Android, Windows, MacOS) and Dart standalone (Windows, MacOS and Linux)
+
+* Flutter ^2.8.0
+
+## Usage
+
+* Add `realm` package to a Flutter application.
+    ```
+    flutter pub add realm
+    ```
+
+* Enable generation of RealmObjects.
+
+    * Add `build_runner` package to `dev_dependencies`.
+        ```
+        dart pub add build_runner --dev
+        ```
+    * Enable `realm_generator` by adding a `build.yaml` file to the application.
+        ```yaml
+        targets:
+            $default:
+                builders:
+                    realm_generator|realm_object_builder:
+                        enabled: true
+                        generate_for:
+                            - lib/*.dart 
+        ```
+
+    * Import Realm in a dart file (ex. `catalog.dart`).
+        ```dart
+        import 'package:realm/realm.dart';
+        ```
+
+    * Declare a part file `catalog.g.dart` in the begining of the `catalog.dart` dart file after all imports.
+
+        ```dart
+        import 'dart:io';
+
+        part 'catalog.g.dart'
+        ```
+
+    * Create a data model class
+
+        ```dart
+        @RealmModel()
+        class _Item {
+            @PrimaryKey()
+            late final int id;
+
+            late String name;
+            
+            int price = 42;
+        }
+        ```
+    * Generate RealmObject class `Item` from data model class `_Item`.
+
+        ```
+        dart run realm generate
+        ```
+        A new file `catalog.g.dart` will be created next to the `catalog.dart`.
+        
+        _*This file should be committed to source control_
+
+    * Use the RealmObject class `Item` with Realm.
+
+        ```dart
+        // Create a Configuration object
+        var config = Configuration(Item.schema);
+
+        // Opean a Realm
+        realm = Realm(config);
+
+        // Open a write transaction
+        realm.write(() {
+          var myItem = Item(0, 'Iteam', price: 4);
```

Comment:
We usually create the objects in our tests before to open transaction. I think it is a good practice, but it could be like this also.

Reply:
it should work regardless. We shouldn't advise on that as a good practice. Doing both is ok